### PR TITLE
chore(lint): bump golangci-lint to v2.13.1 and kube-api-linter + fix new warnings

### PR DESCRIPTION
**Pull request by dongjiang1989**
<!--
Thanks for opening a PR! Please delete any sections that don’t apply.
-->

# Description

This PR bumps the custom golangci-lint build and its plugins, updates the linter configuration to be compatible with the new versions, and fixes all newly-reported warnings across the codebase.                                                                                        
                                                                                                                                                                                                                                                                                         
  ### Dependency bumps                                                                                                                                                                                                                                                                     
  - `golangci-lint`: `v2.9.0` → `v2.13.1`                                                                                                                                                                                                                                                
  - `kube-api-linter`: `v0.0.0-20250715075424-4fab82d26a8e` → `v0.0.0-20260518104151-5ebe05f9440b`                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                           
  ### Linter config changes (`.golangci.yaml`)                                                                                                                                                                                                                                             
  - **`kubeapilinter`**: replaced the wildcard `enable: ["*"]` with an explicit list of enabled sub-linters and `disable: ["*"]`. The wildcard approach now fails validation because several sub-linters (e.g. `conflictingmarkers`, `dependenttags`, `forbiddenmarkers`,                  
  `namingconventions`, `preferredmarkers`) require non-empty configuration. Each enabled sub-linter that requires config now provides one.                                                                                                                                                 
  - **`gomodguard` → `gomodguard_v2`**: `gomodguard` is deprecated since golangci-lint v2.12.0 and replaced by `gomodguard_v2`, which uses a flatter YAML structure (`blocked` is a list of modules rather than a `modules:` map).                                                       
  - **gosec test exclusions**: added `exclusions.rules` to silence noisy gosec rules in test code:                                                                                                                                                                                         
    - `G101` (hardcoded credentials), `G703`/`G704`/`G705` (taint analysis) excluded in `_test.go` files                                                                                                                                                                                   
    - `G101`, `G118` (unclosed context), `G703`/`G704`/`G705` excluded under `test/` directories  

<!--
A concise explanation of the change. You may include:
- **Motivation:** why this change is needed
- **What changed:** key implementation details
- **Related issues:** e.g., `Fixes #123`
-->

# Change Type

<!--
Select one or more of the following by including the corresponding slash-command:
```
/kind breaking_change
/kind bump
/kind cleanup
/kind design
/kind deprecation
/kind documentation
/kind feature
/kind fix
/kind flake
/kind install
```
-->

/kind cleanup                                                                                                                                                                                                                                                                            

/kind bump  

# Changelog

<!--
Provide the exact line to appear in release notes for the chosen changelog type.

If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
-->

```release-note
NONE
```

# Additional Notes

<!--
Any extra context or edge cases for reviewers.
-->

  - Ran make kind-build-and-load / make verify locally; all unit tests pass.                                                                                                                                                                                                               
  - Lint output after the fix: 0 issues.

### CODE REVIEW

File: pkg/pluginsdk/ir/model.go [hunk 1]
```diff
@@ -146,7 +147,7 @@ func NewEndpointsForBackend(us BackendObjectIR) *EndpointsForBackend {
 		h.Write([]byte(k + "=" + v))
 	}
 	h.Write([]byte{0})
-	h.Write([]byte{byte(us.TrafficDistribution)})
+	h.Write([]byte(strconv.Itoa(int(us.TrafficDistribution))))
```
Comment (puertomontt) [inline pkg/pluginsdk/ir/model.go]: this would change the hash

**Review by puertomontt — CHANGES_REQUESTED** [review]

File: pkg/pluginsdk/ir/model.go (hunk 1 above)
Comment (dongjiang1989) [reply pkg/pluginsdk/ir/model.go]: Thanks @puertomontt 
Chang to `slices.Sorted(maps.Keys(labels))`

**Comment by dongjiang1989** [comment]
Thanks @puertomontt Please re-check it

**Comment by puertomontt** [comment]
@dongjiang1989 can you resolve the merge conflict, thanks

**Comment by dongjiang1989** [comment]
> @dongjiang1989 can you resolve the merge conflict, thanks

@puertomontt  Fixed and rebase done.

**Comment by dongjiang1989** [comment]
@puertomontt rebase and update latest golangci-linter version

File: pkg/kgateway/extensions2/plugins/trafficpolicy/transformation_plugin.go [hunk 2]
```diff
@@ -85,9 +85,8 @@ func toRustFormationPerRouteConfig(t *kgateway.TransformationPolicy) (*dynamicmo
 		DynamicModuleConfig: &extensiondynamicmodulev3.DynamicModuleConfig{
 			Name: RustformationModuleName,
 		},
-		FilterName:         RustformationFilterName,
-		PerRouteConfigName: RustformationFilterName,
-		FilterConfig:       filterCfg,
+		FilterName:   RustformationFilterName,
+		FilterConfig: filterCfg,
```
Comment (puertomontt) [inline pkg/kgateway/extensions2/plugins/trafficpolicy/transformation_plugin.go:89]: why did this change, bad merge?

File: .golangci.yaml [hunk 3]
```diff
@@ -56,16 +56,55 @@ linters:
         settings:
           linters:
             enable:
-              - "*"
```
Comment (puertomontt) [inline .golangci.yaml:59]: are there any linters that are missing now?
